src/signature.c

```diff
@@ -157,6 +158,63 @@ GBytes *cms_sign(GBytes *content, const gchar *certfile, const gchar *keyfile, G
 	return res;
 }
 
+gchar* get_pubkey_hash(X509 *cert) {
+		gchar *data = NULL;
+		BIO *bio;
+		unsigned char *ibuf = NULL;
+		unsigned char *buf2;
+		unsigned int len = 0;
+		unsigned int n = 0;
+		unsigned char md[SHA256_DIGEST_LENGTH];
+
+		bio = BIO_new(BIO_s_mem());
+
+		len = i2d_X509_PUBKEY(X509_get_X509_PUBKEY(cert), NULL);
+		buf2 = ibuf = g_malloc(len);
+		if (ibuf) i2d_X509_PUBKEY(X509_get_X509_PUBKEY(cert), &buf2);
```

Comment:
g_malloc only returns NULL if len is 0. We shouldn't even try in that case.

Reply:
Ok, will test for `len <= 0` before and remove the ibuf check